# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/test/unit/head_trr.js
# Commit: 8e4f927ad236
# Full Hash: 8e4f927ad236e3f39665ae7aca84783989a9aaa2
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2020-09-15 15:47:19
# Regressor Bug: 1645108
# File Overlap Count: 2
# Description:
#   Bug 1645108 - Parse additional section of TRR response r=dragana,necko-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D87088
# ==============================================================================

diff -r 8342bb37c478 -r 8e4f927ad236 netwerk/test/unit/head_trr.js
--- a/netwerk/test/unit/head_trr.js	Tue Sep 15 10:32:29 2020 +0000
+++ b/netwerk/test/unit/head_trr.js	Tue Sep 15 10:38:27 2020 +0000
@@ -147,6 +147,7 @@
       this.expectedAnswer,
       `Checking result for ${this.name}`
     );
+    inRecord.rewind(); // In case the caller also checks the addresses
 
     if (this.delay !== undefined) {
       Assert.greaterOrEqual(
@@ -268,17 +269,18 @@
 
   function processRequest(req, resp, payload) {
     let dnsQuery = global.dnsPacket.decode(payload);
-    let answers =
+    let response =
       global.dns_query_answers[
         `${dnsQuery.questions[0].name}/${dnsQuery.questions[0].type}`
-      ] || [];
+      ] || {};
 
     let buf = global.dnsPacket.encode({
       type: "response",
       id: dnsQuery.id,
       flags: global.dnsPacket.RECURSION_DESIRED,
       questions: dnsQuery.questions,
-      answers,
+      answers: response.answers || [],
+      additionals: response.additionals || [],
     });
 
     resp.setHeader("Content-Length", buf.length);
@@ -333,10 +335,11 @@
   ///          flush: false,
   ///          data: "1.2.3.4",
   ///        }]
-  async registerDoHAnswers(name, type, answers) {
-    let text = `global.dns_query_answers["${name}/${type}"] = ${JSON.stringify(
-      answers
-    )}`;
+  async registerDoHAnswers(name, type, answers, additionals) {
+    let text = `global.dns_query_answers["${name}/${type}"] = ${JSON.stringify({
+      answers,
+      additionals,
+    })}`;
     return this.execute(text);
   }
 }